### PR TITLE
[ADAM-2062] Update Spark version to 2.3.2

### LINKED ISSUE #2062: Support Spark 2.3.2

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #2062.

### CODE REVIEW

**Comment by heuermh** [comment]
Works ok with `--packages org.apache.parquet:parquet-avro:1.8.3`
```
$ ./bin/adam-shell --master yarn --packages org.apache.parquet:parquet-avro:1.8.3
Using SPARK_SHELL=/opt/cloudera/parcels/SPARK2-2.2.0.cloudera1-1.cdh5.12.0.p0.142354/lib/spark2//bin/spark-shell
...
org.apache.parquet#parquet-avro added as a dependency
:: resolving dependencies :: org.apache.spark#spark-submit-parent;1.0
	confs: [default]
	found org.apache.parquet#parquet-avro;1.8.3 in central
	found org.apache.parquet#parquet-column;1.8.3 in local-m2-cache
	found org.apache.parquet#parquet-common;1.8.3 in local-m2-cache
	found org.slf4j#slf4j-api;1.7.5 in central
	found org.apache.parquet#parquet-encoding;1.8.3 in local-m2-cache
	found commons-codec#commons-codec;1.5 in local-m2-cache
	found org.apache.parquet#parquet-hadoop;1.8.3 in local-m2-cache
	found org.apache.parquet#parquet-format;2.3.1 in local-m2-cache
	found org.apache.parquet#parquet-jackson;1.8.3 in local-m2-cache
	found org.codehaus.jackson#jackson-mapper-asl;1.9.11 in local-m2-cache
	found org.codehaus.jackson#jackson-core-asl;1.9.11 in local-m2-cache
	found org.xerial.snappy#snappy-java;1.1.1.6 in local-m2-cache
	found org.apache.avro#avro;1.8.0 in local-m2-cache
	found org.codehaus.jackson#jackson-core-asl;1.9.13 in local-m2-cache
	found org.codehaus.jackson#jackson-mapper-asl;1.9.13 in local-m2-cache
	found com.thoughtworks.paranamer#paranamer;2.7 in local-m2-cache
	found org.apache.commons#commons-compress;1.8.1 in local-m2-cache
	found org.tukaani#xz;1.5 in local-m2-cache
	found org.slf4j#slf4j-api;1.7.7 in central
	found it.unimi.dsi#fastutil;6.5.7 in central
downloading https://repo1.maven.org/maven2/org/apache/parquet/parquet-avro/1.8.3/parquet-avro-1.8.3.jar ...
	[SUCCESSFUL ] org.apache.parquet#parquet-avro;1.8.3!parquet-avro.jar (27ms)
downloading .m2/repository/org/apache/parquet/parquet-column/1.8.3/parquet-column-1.8.3.jar ...
	[SUCCESSFUL ] org.apache.parquet#parquet-column;1.8.3!parquet-column.jar (2ms)
downloading .m2/repository/org/apache/parquet/parquet-hadoop/1.8.3/parquet-hadoop-1.8.3.jar ...
	[SUCCESSFUL ] org.apache.parquet#parquet-hadoop;1.8.3!parquet-hadoop.jar (2ms)
downloading .m2/repository/org/apache/parquet/parquet-common/1.8.3/parquet-common-1.8.3.jar ...
	[SUCCESSFUL ] org.apache.parquet#parquet-common;1.8.3!parquet-common.jar (1ms)
downloading .m2/repository/org/apache/parquet/parquet-encoding/1.8.3/parquet-encoding-1.8.3.jar ...
	[SUCCESSFUL ] org.apache.parquet#parquet-encoding;1.8.3!parquet-encoding.jar (1ms)
downloading .m2/repository/org/apache/parquet/parquet-jackson/1.8.3/parquet-jackson-1.8.3.jar ...
	[SUCCESSFUL ] org.apache.parquet#parquet-jackson;1.8.3!parquet-jackson.jar (2ms)
:: resolution report :: resolve 1905ms :: artifacts dl 49ms
	:: modules in use:
	com.thoughtworks.paranamer#paranamer;2.7 from local-m2-cache in [default]
	commons-codec#commons-codec;1.5 from local-m2-cache in [default]
	it.unimi.dsi#fastutil;6.5.7 from central in [default]
	org.apache.avro#avro;1.8.0 from local-m2-cache in [default]
	org.apache.commons#commons-compress;1.8.1 from local-m2-cache in [default]
	org.apache.parquet#parquet-avro;1.8.3 from central in [default]
	org.apache.parquet#parquet-column;1.8.3 from local-m2-cache in [default]
	org.apache.parquet#parquet-common;1.8.3 from local-m2-cache in [default]
	org.apache.parquet#parquet-encoding;1.8.3 from local-m2-cache in [default]
	org.apache.parquet#parquet-format;2.3.1 from local-m2-cache in [default]
	org.apache.parquet#parquet-hadoop;1.8.3 from local-m2-cache in [default]
	org.apache.parquet#parquet-jackson;1.8.3 from local-m2-cache in [default]
	org.codehaus.jackson#jackson-core-asl;1.9.13 from local-m2-cache in [default]
	org.codehaus.jackson#jackson-mapper-asl;1.9.13 from local-m2-cache in [default]
	org.slf4j#slf4j-api;1.7.7 from central in [default]
	org.tukaani#xz;1.5 from local-m2-cache in [default]
	org.xerial.snappy#snappy-java;1.1.1.6 from local-m2-cache in [default]
	:: evicted modules:
	org.slf4j#slf4j-api;1.7.5 by [org.slf4j#slf4j-api;1.7.7] in [default]
	org.codehaus.jackson#jackson-mapper-asl;1.9.11 by [org.codehaus.jackson#jackson-mapper-asl;1.9.13] in [default]
	org.codehaus.jackson#jackson-core-asl;1.9.11 by [org.codehaus.jackson#jackson-core-asl;1.9.13] in [default]
	org.xerial.snappy#snappy-java;1.1.1.3 by [org.xerial.snappy#snappy-java;1.1.1.6] in [default]
	---------------------------------------------------------------------
	|                  |            modules            ||   artifacts   |
	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
	---------------------------------------------------------------------
	|      default     |   21  |   6   |   6   |   4   ||   17  |   6   |
	---------------------------------------------------------------------
:: retrieving :: org.apache.spark#spark-submit-parent
	confs: [default]
	6 artifacts copied, 11 already retrieved (2652kB/19ms)

Spark context available as 'sc' (master = yarn, app id = application_1537216002452_0015).
Spark session available as 'spark'.
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 2.2.0.cloudera1
      /_/

Using Scala version 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_74)
Type in expressions to have them evaluated.
Type :help for more information.

scala> import org.bdgenomics.adam.rdd.ADAMContext._
import org.bdgenomics.adam.rdd.ADAMContext._

scala> val alignments = sc.loadParquetAlignments("foo.alignments.adam")
alignments: org.bdgenomics.adam.rdd.read.AlignmentRecordRDD = ParquetUnboundAlignmentRecordRDD with 25 reference sequences, 1 read groups, and 1 processing steps

scala> alignments.dataset.count()
res0: Long = 199543

scala> alignments.saveAsParquet("foo.alignments.copy.adam")

scala> sc.loadAlignments("foo.alignments.copy.adam").rdd.count()
res2: Long = 199543
```

**Comment by akmorrow13** [comment]
This seems related to https://github.com/bigdatagenomics/adam/issues/1742. Is the current solution to still include` --packages org.apache.parquet:parquet-avro:1.8.3`?

File: pom.xml [hunk 1]
```diff
@@ -18,11 +18,11 @@
...
   <properties>
     <java.version>1.8</java.version>
-    <avro.version>1.8.1</avro.version>
+    <avro.version>1.8.2</avro.version>
     <scala.version>2.11.12</scala.version>
     <scala.version.prefix>2.11</scala.version.prefix>
-    <spark.version>2.3.0</spark.version>
-    <parquet.version>1.8.2</parquet.version>
+    <spark.version>2.3.2</spark.version>
+    <parquet.version>1.8.3</parquet.version>
```
Comment (akmorrow13) [inline pom.xml]: Can we still not remove the pom exclusions for parquet in this update?

Comment (heuermh) [reply pom.xml]: That is correct.  Without the pom exclusions we run into the same `org.apache.avro.SchemaParseException: Can't redefine: list` in unit tests

Comment (akmorrow13) [reply pom.xml]: That's too bad. For some reason, the --packages does not work for parquet-hadoop in python.

Comment (heuermh) [reply pom.xml]: Unfortunately, I don't understand why it is necessary for running on CDH Spark and not standalone or EMR, or why it works.

Comment (akmorrow13) [reply pom.xml]: I have gone through all possible combinations of potential answers to the python problem, and the only thing that seems to work is setting:
`--conf spark.driver.extraClassPath=/<path_to_jar>/org.apache.parquet_parquet-hadoop-1.8.3.jar`

for pyspark. This means the user has to download the jar first somewhere. I cannot find a cleaner solution.

**Comment by akmorrow13** [comment]
Thanks @heuermh !

**Pull request merged**